tasks/setup-debian.yml

```diff
@@ -1,26 +1,28 @@
 ---
-- name: Update APT package cache
-  apt:
-    update_cache: true
-    cache_valid_time: 3600
-  when: ansible_distribution == "Ubuntu"
+- name: Setup WireGuard preference
+  copy:
+    src: etc/apt/preferences.d/limit-unstable
+    dest: /etc/apt/preferences.d/limit-unstable
+    owner: root
+    group: root
+    mode: 0644
   tags:
     - wg-install
 
-- name: Install required packages
-  package:
-    name: "{{ item }}"
+- name: Add WireGuard key
+  apt_key:
+    keyserver: "keyserver.ubuntu.com"
+    id: "8B48AD6246925553"
     state: present
-  with_items:
-    - software-properties-common
-    - linux-headers-{{ ansible_kernel }}
+  run_once: true
   tags:
     - wg-install
 
 - name: Add WireGuard repository
   apt_repository:
-    repo: "ppa:wireguard/wireguard"
+    repo: "deb http://deb.debian.org/debian/ unstable main"
     state: present
     update_cache: yes
+  run_once: true
```

Comment:
Copied this from the issue #5. Do not think it is needed. Could fix this and squash if you prefer this